src/main.cpp

```diff
@@ -8156,36 +8153,30 @@ bool SendMessages(CNode* pto, bool fSendTrickle)
 
 void IncrementCurrentNeuralNetworkSupermajority(std::string NeuralHash, std::string GRCAddress, double distance)
 {
-    if (NeuralHash.length() < 5) return;
-    double temp_hashcount = 0;
-    if (mvCurrentNeuralNetworkHash.size() > 0)
-    {
-            temp_hashcount = mvCurrentNeuralNetworkHash[NeuralHash];
-    }
+    if (NeuralHash.length() < 5)
+        return;    
+
     // 6-13-2015 ONLY Count Each Neural Hash Once per GRC address / CPID (1 VOTE PER RESEARCHER)
-    std::string Security = ReadCache("currentneuralsecurity",GRCAddress).value;
+    const std::string& Security = ReadCache("currentneuralsecurity",GRCAddress).value;
     if (Security == NeuralHash)
     {
         //This node has already voted, throw away the vote
         return;
     }
+    
     WriteCache("currentneuralsecurity",GRCAddress,NeuralHash,GetAdjustedTime());
-    if (temp_hashcount == 0)
-    {
-        mvCurrentNeuralNetworkHash.insert(map<std::string,double>::value_type(NeuralHash,0));
-    }
-    double multiplier = 200;
-    if (distance < 40) multiplier = 400;
+
+    double multiplier = distance < 40 ? 400 : 200;
     double votes = (1/distance)*multiplier;
-    temp_hashcount += votes;
-    mvCurrentNeuralNetworkHash[NeuralHash] = temp_hashcount;
+    double& hashcount = mvCurrentNeuralNetworkHash[NeuralHash];
```

Comment:
No need to extract the hashCount here. It's good enough to do `mvCurrentNeuralNetworkHash[NeuralHash] += votes`.